src/main/java/com/ibm/eventstreams/connect/jdbcsink/database/writer/JDBCWriter.java

```diff
@@ -50,107 +51,205 @@ private boolean doesTableExist(Connection connection, String tableName) throws S
         return table.next();
     }
 
-    // TODO: verify will work for all database flavors
-    private String getDatabaseFieldType(Schema.Type type) {
-        switch (type) {
+    private boolean isPostgreSQL(Connection connection) throws SQLException {
+        return connection.getMetaData().getDatabaseProductName().toLowerCase().contains("postgresql");
+    }
+
+    private boolean isDB2(Connection connection) throws SQLException {
+        return connection.getMetaData().getDatabaseProductName().toLowerCase().contains("db2");
+    }
+
+    private boolean isMySQL(Connection connection) throws SQLException {
+        return connection.getMetaData().getDatabaseProductName().toLowerCase().contains("mysql");
+    }
```

Comment:
Does MYSQL work in this one?

Reply:
Yes, we ran this connector in MySQL and DB2, I believe it was also run in Postgres by someone as well.